src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const finalHistoryArray = [];
+  let modifiedObject = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === 'clear') {
+      modifiedObject = {};
+
+      finalHistoryArray.push({ ...modifiedObject });
+    } else if (action.type === 'addProperties') {
+      Object.assign(modifiedObject, action.extraData);
+      finalHistoryArray.push({ ...modifiedObject });
+    } else if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((key) => delete modifiedObject[key]);
+      finalHistoryArray.push({ ...modifiedObject });
+    }
```

Comment:
Good work but let's rewrite it to switch/case statement since we rely on a single action.type value